types/state.go

```diff
@@ -112,6 +112,9 @@ func (s *State) GetChunks(stream *ConfiguredStream) *Set[Chunk] {
 
 // set chunks
 func (s *State) SetChunks(stream *ConfiguredStream, chunks *Set[Chunk]) {
+	if stream.Self().Stream.SyncMode != CDC {
```

Comment:
we must add it for full refresh as we may introduce other types in future
```suggestion
	if stream.Self().Stream.SyncMode != FULL_REFRESH {
```